Release/v3.3 (GIT8266O-490)

bonnefoi opened the pull request:
Could you add the following lines in the ```esp_attr.h```:
```
// Forces data into noinit section to avoid initialization after restart.
#define __NOINIT_ATTR _SECTION_ATTR_IMPL(".noinit", __COUNTER__)

// Forces data into RTC slow memory of .noinit section.
// Any variable marked with this attribute will keep its value
// after restart or during a deep sleep / wake cycle.
#define RTC_NOINIT_ATTR  _SECTION_ATTR_IMPL(".rtc_noinit", __COUNTER__)
```
This modification is necessary to use RTC_NOINIT_ATTR.

donghengqaz committed:
fix(lib): fix probe request length error

donghengqaz committed:
Merge branch 'bugfix/fix_probe_request_length_error_v3.3' into 'release/v3.3'

fix(lib): fix probe request length error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1216

donghengqaz committed:
feat(util): refactor SHA algorithm of util components

donghengqaz committed:
feat(esp-tls): modify esp-tls structure members' offset of platform dependece

donghengqaz committed:
chore(mqtt): modify default mqtt

IBM MQTT will be removed at v3.4.

donghengqaz committed:
Merge branch 'feature/modify_esp_tls_structure_v3.3' into 'release/v3.3'

esp-tls: modify esp-tls structure members' offset of platform dependece (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1220

donghengqaz committed:
Merge branch 'chore/modify_default_mqtt' into 'release/v3.3'

mqtt: modify default mqtt

See merge request sdk/ESP8266_RTOS_SDK!1219

Commit:
feat(lib): add more info when softap deny station

donghengqaz committed:
feat(esp8266): fix CPU light sleep process

donghengqaz committed:
Merge branch 'feature/refactor_util_sha_algorithm_v3.3' into 'release/v3.3'

util: refactor SHA algorithm of util components (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1223

donghengqaz committed:
Merge branch 'bugfix/fix_cpu_sleep_reject_proc_v3.3' into 'release/v3.3'

esp8266: fix CPU light sleep process (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1226

donghengqaz committed:
Merge branch 'feature/add_more_info_when_softap_deny_station' into 'release/v3.3'

feat(lib): add more info when softap deny station

See merge request sdk/ESP8266_RTOS_SDK!1224

cwespressif committed:
fix(lib): fix issue when scanning wpa2_enterprise authmode routers

krzychb committed:
doc: Limit sphinxcontrib versions to <2.0.0 as we use Sphinx 1.8.5

Commit:
feat(esp8266): set wifi default rx continuous buffer to 8

donghengqaz committed:
Merge branch 'feature/set_wifi_default_rx_continuous_buffer_to_8_v3.3' into 'release/v3.3'

feat(esp8266): set wifi default rx continuous buffer to 8 (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1232

donghengqaz committed:
Merge branch 'bugfix/fix_issue_when_scanning_wpa2_ent_routes_v3.3' into 'release/v3.3'

fix(lib): fix issue when scanning wpa2_enterprise authmode routers (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1229

donghengqaz committed:
fix(util): fix sha384/512 data calculation overflow

donghengqaz committed:
Merge branch 'bugfix/fix_sha_data_overflow_v3.3' into 'release/v3.3'

util: fix sha384/512 data calculation overflow (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1234

donghengqaz committed:
feat(mbedtls): mbedtls using SHA/AES of util component instead of its raw algorithm

Commit:
bugfix(flash): add support for GD25Q64 qio enable

donghengqaz committed:
fix(esp-wolfssl): fix esp-wolfssl cmake compiling error

donghengqaz committed:
Merge branch 'feature/mbedtls_using_util_algorithm_v3.3' into 'release/v3.3'

mbedtls: mbedtls using SHA/AES of util component instead of its raw algorithm (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1240

donghengqaz committed:
Merge branch 'bugfix/support_GD25Q64_qio_enable_v3.3' into 'release/v3.3'

bugfix(flash): add support for GD25Q64 qio enable (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1241

donghengqaz committed:
Merge branch 'bugfix/doc_package_versions_v3.3' into 'release/v3.3'

doc: Limit sphinxcontrib versions to <2.0.0 as we use Sphinx 1.8.5 (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1247

donghengqaz committed:
fix(mbedtls): fix option dependence of "MBEDTLS_SERVER_SSL_SESSION_TICKETS"

MBEDTLS_SERVER_SSL_SESSION_TICKETS
    ->  MBEDTLS_CIPHER_MODE_AEAD
        -> MBEDTLS_GCM_C || MBEDTLS_CCM_C || MBEDTLS_CHACHAPOLY_C

donghengqaz committed:
Merge branch 'bugfix/fix_espwolfssl_cmake_compiling_v3.3' into 'release/v3.3'

esp-wolfssl: fix esp-wolfssl cmake compiling error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1254

donghengqaz committed:
Merge branch 'bugfix/fix_mbedtls_option_dependence_v3.3' into 'release/v3.3'

mbedtls: fix option dependence of "MBEDTLS_SERVER_SSL_SESSION_TICKETS" (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1252

donghengqaz committed:
feat(util): support SHA224/SHA384 calculation although call sha256/sha512 APIs

donghengqaz committed:
fix(esp8266): fix light sleep time error when waking up

donghengqaz committed:
Merge branch 'bugfix/fix_uitl_sha_for_mbedtls_error_v3.3' into 'release/v3.3'

util: support SHA224/SHA384 calculation although call sha256/sha512 APIs (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1261

donghengqaz committed:
fix(esp8266): fix Wi-Fi initialization data error when compiling cpp

Commit:
fix(esp8266): fix compile failed when CONFIG_ESP8266_WIFI_CONNECT_OPEN_ROUTER_WHEN_PWD_IS_SET is unset

donghengqaz committed:
Merge branch 'bugfix/fix_wifi_init_data_compiling_error_by_cpp_v3.3' into 'release/v3.3'

esp8266: fix Wi-Fi initialization data error when compiling cpp (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1264

donghengqaz committed:
Merge branch 'bugfix/fix_light_sleep_time_error_when_wakup_v3.3' into 'release/v3.3'

esp8266: fix light sleep time error when waking up (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1266

donghengqaz committed:
Merge branch 'bugfix/fix_compile_failed_when_CONFIG_ESP8266_WIFI_CONNECT_OPEN_ROUTER_WHEN_PWD_IS_SET_is_unset_v3.3' into 'release/v3.3'

fix(esp8266): fix compile failed when CONFIG_ESP8266_WIFI_CONNECT_OPEN_ROUTER_WHEN_PWD_IS_SET is unset (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1268

donghengqaz committed:
fix(esp-mqtt): fix esp-mqtt API declare error

donghengqaz committed:
Merge branch 'bugfix/fix_mqtt_api_declare_error_v3.3' into 'release/v3.3'

esp-mqtt: fix esp-mqtt API declare error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1270

ESP-YJM committed:
feat(ipv6): Enable ipv6 autoconfig function in lwip port level

donghengqaz committed:
feat(newlib): add "sys/un.h" header file

donghengqaz committed:
Merge branch 'feature/add_header_file_un_h_v3.3' into 'release/v3.3'

newlib: add "sys/un.h" header file (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1272

donghengqaz committed:
feat(tcpip_adapter): add header file and wrapper APIs to make code compatible with "esp_netif"

donghengqaz committed:
Merge branch 'feature/add_apis_to_compatible_with_esp_netif_v3.3' into 'release/v3.3'

tcpip_adapter: add header file and wrapper APIs to make code compatible with "esp_netif" (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1275

Commit:
bugfix(uart_interrupt_occupy):disable intr that you are not using

supreetd21 committed:
bugfix/tls_handshake: Fixed an error check during TLS handshake.

(cherry picked from commit fb8b45cf0665d7cca1415430a22c1d74d17d9827)

Commit:
bugfix(gpio_set_pull_mode_error):fix cant set gpio pull mode

donghengqaz committed:
Merge branch 'bugfix/tls_handshake_backport_v3.3' into 'release/v3.3'

bugfix/tls_handshake:(Backport v3.3) Fixed an error check during TLS handshake.

See merge request sdk/ESP8266_RTOS_SDK!1290

donghengqaz committed:
Merge branch 'feature/add_enable_ipv6_autoconfig' into 'release/v3.3'

feat(ipv6): Enable ipv6 autoconfig function in lport level

See merge request sdk/ESP8266_RTOS_SDK!1281

donghengqaz committed:
Merge branch 'bugfix/gpio_set_pull_mode_error_v3.3' into 'release/v3.3'

bugfix(gpio_set_pull_mode_error):fix cant set gpio pull mode (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1287

donghengqaz committed:
fix(example): fix HTTPS request SSL macro error

Commit:
fix(requirements): fix read the docs compiling requirements error

Commit:
fix(esp-tls): fix esp-tls compiling error

Closes https://github.com/espressif/ESP8266_RTOS_SDK/issues/828

donghengqaz committed:
Merge branch 'bugfix/fix_read_the_docs_requires_v3.3' into 'release/v3.3'

fix(requirements): fix read the docs compiling requirements error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1301

donghengqaz committed:
Merge branch 'bugfix/fix_esp_tls_server_compiling_error_v3.3' into 'release/v3.3'

fix(esp-tls): fix esp-tls compiling error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1304

donghengqaz committed:
Merge branch 'bugfix/fix_example_https_request_ssl_v3.3' into 'release/v3.3'

fix(example): fix HTTPS request SSL macro error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1307

donghengqaz committed:
Merge branch 'bugfix/uart_interrupt_occupy_v3.3' into 'release/v3.3'

bugfix(uart_interrupt_occupy):disable intr that you are not using(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1279

Commit:
fix(esp8266): fix deep sleep fail

donghengqaz committed:
Merge branch 'bugfix/fix_deep_sleep_fail_v3.3' into 'release/v3.3'

fix(esp8266): fix deep sleep fail (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1309

donghengqaz committed:
chore(ci): modify CI test SSC branch to be "release/v3.3_8266" for backport

donghengqaz committed:
Merge branch 'feature/modify_ci_test_ssc_branch_for_backport' into 'release/v3.3'

chore(ci): modify CI test SSC branch to be "release/v3.3_8266" for backport

See merge request sdk/ESP8266_RTOS_SDK!1316

Commit:
fix(wifi): fix writting error

donghengqaz committed:
Merge branch 'bugfix/fix_writting_error_v3_3' into 'release/v3.3'

fix(wifi): fix writting error

See merge request sdk/ESP8266_RTOS_SDK!1324

donghengqaz committed:
feat(bootloader): add option to disable JTAG I/O when boot

donghengqaz committed:
fix(lib): fix sleep error when period cal called

donghengqaz committed:
fix(lib): do not connect to ap when set country called

donghengqaz committed:
Merge branch 'bugfix/fix_sleep_error_when_period_cal_called_v3.3' into 'release/v3.3'

fix(lib): fix sleep error when period cal called (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1334

donghengqaz committed:
Merge branch 'bugfix/do_not_connect_to_ap_when_set_country_called_v3.3' into 'release/v3.3'

fix(lib): do not connect to ap when set country called (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1335

donghengqaz committed:
feat(tcpip_adapter): add AP assigned IP event data

donghengqaz committed:
Merge branch 'feature/add_ap_sta_ip_assigned' into 'release/v3.3'

feat(tcpip_adapter): add AP assigned IP event data

See merge request sdk/ESP8266_RTOS_SDK!1336

donghengqaz committed:
Merge branch 'bugfix/uart_cannot_write_completely' into 'master'

fix: fix uart write data incomplete

See merge request sdk/ESP8266_RTOS_SDK!1276

(cherry picked from commit b0e5afe324fad4fb2ed9a94237fc29e0b10c1ad1)

ae51c20e fix: fix uart write data incomplete

Commit:
feat(pm): refactor power management according to IDF

Commit:
feat(lib): update library

donghengqaz committed:
feat(phy): update phy library from 1155.0 to 1156.0

1. fix bug in calibration
2. reduce RAM cost
3. reduce current when RF is closed

donghengqaz committed:
Merge branch 'feature/update_phy_library_v3.3' into 'release/v3.3'

feat(phy): update phy library from 1155.0 to 1156.0 (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1342

ESP-YJM committed:
feat(lwip): Add kconfig to congigure initial RTO time and set default value to 1000

donghengqaz committed:
Merge branch 'feature/make_tcp_rto_configurable' into 'release/v3.3'

feat(lwip): Add kconfig to congigure initial RTO time and set default value to 1000

See merge request sdk/ESP8266_RTOS_SDK!1344

ESP-YJM committed:
feat(tls): update esp-tls and tcp_transport from idf
Commit ID:88bf21b2

donghengqaz committed:
Merge branch 'feature/refactor_set_channel_and_power_management_according_to_IDF_v3.3' into 'release/v3.3'

feat(pm): refactor power management according to IDF (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1339

ESP-YJM committed:
feat(tls): Modify tls component for building

Commit:
fix(lib): fix memleak when station send illegal packet

donghengqaz committed:
Merge branch 'feature/update_tls_from_idf_v3.3' into 'release/v3.3'

feat(tls): update tls from idf v3.3 (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1347

donghengqaz committed:
Merge branch 'bugfix/fix_memleak_when_sta_send_illegal_pkt_v3.3' into 'release/v3.3'

fix(lib): fix memleak when station send illegal packet

See merge request sdk/ESP8266_RTOS_SDK!1349

cwespressif committed:
fix(wpa2_ent): fix the issue of disable wpa2_enter

donghengqaz committed:
Merge branch 'bugfix/fix_the_issue_of_disable_wpa2_ent' into 'release/v3.3'

fix(wpa2): fix the issue of disable wpa2_enterprise

See merge request sdk/ESP8266_RTOS_SDK!1354

Commit:
feat(lib): refactor deep sleep according to ESP32

donghengqaz committed:
Merge branch 'feature/refactor_deep_sleep_according_to_ESP32_v3.3' into 'release/v3.3'

feat(lib): refactor deep sleep according to ESP32

See merge request sdk/ESP8266_RTOS_SDK!1355

cwespressif committed:
feat(wpa2_ent): add wifi_unregister_wpa2_cb

donghengqaz committed:
Merge branch 'feature/add_wifi_unregister_wpa2_cb' into 'release/v3.3'

feat(wpa2_ent): add wifi_unregister_wpa2_cb

See merge request sdk/ESP8266_RTOS_SDK!1358

donghengqaz committed:
feat(lwip): add shutdown to socket_mt

donghengqaz committed:
fix(lwip): fix IGMP join wrong address

Commit:
fix(lib): fix return code error when set wrong authmode in softap config

donghengqaz committed:
Merge branch 'bugfix/fix_return_code_error_when_set_wrong_authmode_in_softap_config_v3.3' into 'release/v3.3'

fix(lib): fix return code error when set wrong authmode in softap config(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1367

donghengqaz committed:
Merge branch 'bugfix/fix_ipv6_igmp_error' into 'release/v3.3'

fix(lwip): fix IGMP join wrong address

See merge request sdk/ESP8266_RTOS_SDK!1366

donghengqaz committed:
Merge branch 'feature/add_shutdown_to_socket_mt' into 'release/v3.3'

feat(lwip): add shutdown to socket_mt

See merge request sdk/ESP8266_RTOS_SDK!1365

donghengqaz committed:
fix(lwip): fix DHCP rebind timeout has no process

donghengqaz committed:
Merge branch 'bugfix/fix_dhcp_rebind_error_proc_lost' into 'release/v3.3'

fix(lwip): fix DHCP rebind timeout has no process

See merge request sdk/ESP8266_RTOS_SDK!1369

donghengqaz committed:
fix(tcpip_adapter): fix local IP changed when Wi-Fi disconnect

donghengqaz committed:
Merge branch 'bugfix/fix_addr_changed_when_wifi_disconnect_v3.3' into 'release/v3.3'

fix(tcpip_adapter): fix local IP changed when Wi-Fi disconnect (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1377

donghengqaz committed:
feat(phy): update phy library from 1156.0 to 1059.0

1. add phy configuration parameters calibration

donghengqaz committed:
Merge branch 'feature/update_phy_library_v3.3' into 'release/v3.3'

feat(phy): update phy library from 1156.0 to 1059.0 (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1380

donghengqaz committed:
Merge branch 'cherry-pick-b0e5afe3' into 'release/v3.3'

Merge branch 'bugfix/uart_cannot_write_completely' into 'master'

See merge request sdk/ESP8266_RTOS_SDK!1370

donghengqaz committed:
Revert "Merge branch 'bugfix/fix_addr_changed_when_wifi_disconnect_v3.3' into 'release/v3.3'"

This reverts merge request !1377

donghengqaz committed:
Merge branch 'revert-049ebe1d' into 'release/v3.3'

Revert "Merge branch 'bugfix/fix_addr_changed_when_wifi_disconnect_v3.3' into 'release/v3.3'"

See merge request sdk/ESP8266_RTOS_SDK!1387

xiongyumail committed:
bugfix(i2s): remove_i2s_bbpll_clk_enable

* In the latest version, i2c_bbpll is turned on by PHY by default. Configuring the register again may affect the bbpll state.

* Model sleep does not affect the I2S clock. I2S cannot be used under light sleep.

donghengqaz committed:
feat(mbedtls): Update mbedTLS version to v2.16.5

donghengqaz committed:
feat(mbedtls): Add dynamic buffer and local resource managment to decrease SSL heap cost

donghengqaz committed:
feat(esp8266): Panic uses hardware restart instead of software restart

donghengqaz committed:
Merge branch 'feature/panic_using_hardware_restart_instead_of_software_v3.3' into 'release/v3.3'

feat(esp8266): Panic uses hardware restart instead of software restart (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1403

donghengqaz committed:
feat(wifi): Add Wi-Fi send PLCP function

Commit:
fix(wifi): fix writing errors

donghengqaz committed:
Merge branch 'feature/add_wifi_send_plcp_v3.3' into 'release/v3.3'

feat(wifi): Add Wi-Fi send PLCP function (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1407

donghengqaz committed:
Merge branch 'bugfix/fix_writing_errors_v3.3' into 'release/v3.3'

fix(wifi): fix writing errors(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1409

donghengqaz committed:
Merge branch 'feature/update_mbedtls_to_v2.16.5_v3.3' into 'release/v3.3'

feat(mbedtls): Update mbedTLS version to v2.16.5 (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1411

donghengqaz committed:
Merge branch 'feature/mbedtls_dynamic_memory_v3.3' into 'release/v3.3'

feat(mbedtls): Add dynamic buffer and local resource managment to decrease SSL heap cost (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1412

donghengqaz committed:
fix(lwip): Fix "shutdown" blocks "close" when enable multi-thread support

donghengqaz committed:
feat(lwip): Add option to receive rest data although TCP aborts

donghengqaz committed:
Merge branch 'feature/disable_jtag_io_when_boot_v3.3' into 'release/v3.3'

feat(bootloader): add option to disable JTAG I/O when boot (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1423

donghengqaz committed:
Merge branch 'feature/add_option_recv_rst_data_although_tcp_aborts' into 'release/v3.3'

feat(lwip): Add option to receive rest data although TCP aborts

See merge request sdk/ESP8266_RTOS_SDK!1420

Commit:
bugfix(ledc):fix pwm duty setting error

Commit:
fix(lib): fix connect timeout error

donghengqaz committed:
Merge branch 'bugfix/fix_connect_timeout_error_v3.3' into 'release/v3.3'

fix(lib): fix connect timeout error(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1428

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/ESP8266_RTOS_SDK?pullRequest=917) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/espressif/ESP8266_RTOS_SDK?pullRequest=917) before we can accept your contribution.<br/>**0** out of **7** committers have signed the CLA.<br/><br/>:x: Zhang Jun Hao<br/>:x: xiongyumail<br/>:x: donghengqaz<br/>:x: wujiangang<br/>:x: projectgus<br/>:x: cwespressif<br/>:x: ESP-YJM<br/><hr/>**Zhang Jun Hao** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/ESP8266_RTOS_SDK?pullRequest=917) it.</sub>

ESP-YJM committed:
feat(dhcp): Add LWIP DHCP option12, option60 and option61 in DHCP discovery and request state

donghengqaz committed:
Merge branch 'bugfix/fix_shutdown_block_close' into 'release/v3.3'

fix(lwip): Fix "shutdown" blocks "close" when enable multi-thread support

See merge request sdk/ESP8266_RTOS_SDK!1418

donghengqaz committed:
Merge branch 'bugfix/ledc_pwm_error_v3.3' into 'release/v3.3'

bugfix(ledc):fix pwm duty setting error (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1426

donghengqaz committed:
Merge branch 'feature/add_dhcp_option' into 'release/v3.3'

feat(dhcp): Add DHCP option12, option60 and option61 in DHCP discovery and request state

See merge request sdk/ESP8266_RTOS_SDK!1432

donghengqaz committed:
Merge branch 'bugfix/remove_i2s_bbpll_clk_enable_v3.3' into 'release/v3.3'

bugfix(i2s): remove_i2s_bbpll_clk_enable(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1425

Commit:
fix(lib): do not refuse iphone when iphone reconnect to softap

Commit:
fix(lib): fix crash when esp_restart called

donghengqaz committed:
feat(phy): update phy library from 1159.0 to 1163.0

1. optimize TX EVM
2. optimize clock setting in wake-up

donghengqaz committed:
Merge branch 'feature/update_phy_library_v3.3' into 'release/v3.3'

feat(phy): update phy library from 1159.0 to 1163.0

See merge request sdk/ESP8266_RTOS_SDK!1453

donghengqaz committed:
docs(policy): Add SDK support period policy

donghengqaz committed:
Merge branch 'docs/add_support_period_policy_v3.3' into 'release/v3.3'

docs(policy): Add SDK support period policy (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1462

Commit:
fix(lib): drop beacon when authmode changed

donghengqaz committed:
Merge branch 'bugfix/do_not_refuse_iphone_when_iphone_reconnect_to_softap_v3.3' into 'release/v3.3'

fix(lib): do not refuse iphone when iphone reconnect to softap(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1442

donghengqaz committed:
Merge branch 'bugfix/fix_crash_when_esp_restart_called_v3.3' into 'release/v3.3'

fix(lib): fix crash when esp_restart called(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1447

Commit:
feat(example): set authmode threshold to WPA2 when password set

donghengqaz committed:
fix(xtensa): Fix NMI reenter when multi enable MAC interrupt

donghengqaz committed:
Merge branch 'bugfix/fix_nmi_reenter_when_multi_enable_mac_interrupt_v3.3' into 'release/v3.3'

fix(xtensa): Fix NMI reenter when multi enable MAC interrupt (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1469

donghengqaz committed:
Merge branch 'bugfix/drop_beacon_when_authmode_changed_v3.3' into 'release/v3.3'

fix(lib): drop beacon when authmode changed(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1464

donghengqaz committed:
fix(mbedtls): Fix dynamic buffer mbedtls handshake steps process return 0 when handshake fail

Commit:
fix(lib): fix datarate is always 1M when esp8266 as softap mode

donghengqaz committed:
Merge branch 'bugfix/fix_datarate_is_always_1M_when_esp8266_as_softap_mode' into 'release/v3.3'

fix(lib): fix datarate is always 1M when esp8266 as softap mode

See merge request sdk/ESP8266_RTOS_SDK!1476

donghengqaz committed:
fix(mbedtls): Fix mbedtls dynamic resource leaks memory

cwespressif committed:
fix(wps): fix the issues of memory leak which caused by calling esp_wifi_wps_start frequently

donghengqaz committed:
Merge branch 'bugfix/fix_memory_leak_issue_caused_by_wps_start_frequently' into 'release/v3.3'

fix(wps): fix the issues of memory leak which caused by calling esp_wifi_wps_start frequently

See merge request sdk/ESP8266_RTOS_SDK!1492

Commit:
fix(lib): fix softap PMK invalidata in NVS

donghengqaz committed:
Merge branch 'bugfix/fix_softap_PMK_invalidate_in_NVS_v3.3' into 'release/v3.3'

fix(lib): fix softap PMK invalidata in NVS(backport V3.3)

See merge request sdk/ESP8266_RTOS_SDK!1495

donghengqaz committed:
Merge branch 'bugfix/fix_mbedtls_dynamic_mem_leak_heap_v3.3' into 'release/v3.3'

fix(mbedtls): Fix mbedtls dynamic resource leaks memory (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1485

ESP-YJM committed:
fix(tcp): optimize tcp pcb allocation algoritm

Commit:
feat(lib): do not change wifi mode when esp_deep_sleep() called

donghengqaz committed:
Merge branch 'feature/do_not_change_wifi_mode_in_nvs_when_esp_deep_sleep_called_v3.3' into 'release/v3.3'

feat(lib): do not change wifi mode when esp_deep_sleep() called

See merge request sdk/ESP8266_RTOS_SDK!1506

donghengqaz committed:
Merge branch 'bugfix/optimize_tcp_pcb_allocation_algoritm' into 'release/v3.3'

fix(tcp): optimize tcp pcb allocation algoritm

See merge request sdk/ESP8266_RTOS_SDK!1510

wujiangang committed:
feat(tools): Add xtensa-lx106-elf toolchain url

Commit:
fix(lib): do not rewrite flash when set station config

donghengqaz committed:
Merge branch 'bugfix/do_not_rewrite_flash_when_set_sta_config_v3.3' into 'release/v3.3'

fix(lib): do not rewrite flash when set station config

See merge request sdk/ESP8266_RTOS_SDK!1526

Commit:
fix(lib): do not realloc bcn when tx bcn

donghengqaz committed:
Merge branch 'bugfix/do_not_realloc_bcn_when_tx_bcn_v3.3' into 'release/v3.3'

fix(lib): do not realloc bcn when tx bcn

See merge request sdk/ESP8266_RTOS_SDK!1530

donghengqaz committed:
Merge branch 'feature/add_xtensa-lx106-elf_toolchain_url_v3.3' into 'release/v3.3'

feat(tools): Add xtensa-lx106-elf toolchain url (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1536

donghengqaz committed:
Merge branch 'bugfix/fix_dynamic_mbedtls_return_0_when_handshake_fail_v3.3' into 'release/v3.3'

fix(mbedtls): Fix dynamic buffer mbedtls handshake steps process return 0 when handshake fail (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1475

Commit:
fix(freertos): disable other task before disable nmi

donghengqaz committed:
Merge branch 'bugfix/disable_other_task_before_nmi_disabled_v3.3' into 'release/v3.3'

fix(freertos): disable other task before disable nmi(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1570

donghengqaz committed:
fix(esp8266): Fix cmake not compile and download phy_init_data.bin

donghengqaz committed:
Merge branch 'bugfix/fix_cmake_enable_phy_init_bin_partition_err_v3.3' into 'release/v3.3'

fix(esp8266): Fix cmake not compile and download phy_init_data.bin (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1575

Commit:
fix(lib): fix some bugs
 - fix connect error when ap g rate in xrate ie
 - fix drop eapol key when retry bit is set
 - drop pkt that hw dump error
 - fix block scan timeout when user set time large than 5s

donghengqaz committed:
Merge branch 'bugfix/fix_block_sacn_and_some_JAP_issues_v3.3' into 'release/v3.3'

fix(lib): fix some bugs

See merge request sdk/ESP8266_RTOS_SDK!1576

Commit:
fix(lib): fix i2c error when calling esp_reastart()

donghengqaz committed:
Merge branch 'bugfix/fix_i2c_error_when_esp_restart_called_v3.3' into 'release/v3.3'

fix(lib): fix i2c error when calling esp_reastart()

See merge request sdk/ESP8266_RTOS_SDK!1590

ESP-YJM committed:
fix(mbedtls): Fix mbedtls_ssl_send_alert_message crash due to ssl->out_iv is NULL

donghengqaz committed:
Merge branch 'bugfix/fix_mbedtls_send_alert_crash_v3.3' into 'release/v3.3'

fix(mbedtls): Fix mbedtls_ssl_send_alert_message crash due to ssl->out_iv is NULL(backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1604

projectgus committed:
freertos: Add queue init overflow check

Based on FreeRTOS kernel patch 47338393 but modified
to work without assertions.

projectgus committed:
freertos: Add addition overflow check for stream buffer

Patch from upstream commit d05b9c123f2bf9090bce386a244fc934ae44db5b

donghengqaz committed:
Merge branch 'bugfix/queue_arith_overflow_v3.3' into 'release/v3.3'

freertos: Add queue init overflow check (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1612

donghengqaz committed:
Tools: Constrain the cryptography package for avoiding breaking changes

The cryptography 35 release causes breaking changes. Some of
certificates are rendered invalid and the package cannot be build
without rust. See
https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst#3500---2021-09-29
for more details.

donghengqaz committed:
Merge branch 'bugfix/constrain_cryptography_v3.3' into 'release/v3.3'

Tools: Constrain the cryptography package for avoiding breaking changes (backport v3.3)

See merge request sdk/ESP8266_RTOS_SDK!1642
